src/QMCHamiltonians/LocalMomentEstimator.cpp

```diff
@@ -20,7 +20,7 @@
 namespace qmcplusplus
 {
 LocalMomentEstimator::LocalMomentEstimator(ParticleSet& elns, ParticleSet& srcs)
-  : ions(srcs), d_table_ID(elns.addTable(ions, DT_AOS))
+  : ions(srcs), d_table_ID(elns.addTable(srcs, DT_AOS))
```

Comment:
This looks consequential. Are there any tests on this code?

Reply:
We don't have any test for this class. I have no idea what this class does.
The initial code was
```
LocalMomentEstimator::LocalMomentEstimator(ParticleSet& elns, ParticleSet& srcs)
  : ions(srcs)
{
  d_table_ID = elns.addTable(ions, DT_AOS);
}
```
I change the code to 
```
LocalMomentEstimator::LocalMomentEstimator(ParticleSet& elns, ParticleSet& srcs)
  : ions(srcs), d_table_ID(elns.addTable(ions, DT_AOS))
{
}
```
This was wrong because the order of initialization list doesn't affect the actual order of initialization which is determined by the order of declaration. Clang reports warn of using uninitialized objects.

In the future, we need to turn on compiler warning or even error all unused and uninitialized variables.

Reply:
Paging @jtkrogel

Reply:
It is in the manual. Code is clearly AoS only. Maybe there is a test (I can't currently check)

Best not to change the code, assuming the old way compiles.

Once the new nightlies have been stabilized we will add a build with more warnings and checking enabled.

Reply:
There is a test in tests/estimator/latdev. pandas has to be installed for it to be enabled & run.

Since Jaron is out I suggest simply not modifying this code.

Reply:
@prckent I can't get your point for not fixing this. The warning is accurate and surfaces a bug I just introduced a few days ago. I hit similar issues when I was cleaning up addTable.
I made a test by manually adding the xml entry to activate this estimator and got a segfault which disappeared after this fix. I think we need this PR for solving the segfault. I have no idea what this estimator exactly does and hence can't construct a real test to check the result numbers.

What did you mean "in the manual" and "tests/estimator/latdev"? I didn't find any manual section or test files correspond to this estimator particularly.

Reply:
I was looking at the latticedeviation and not localmoment estimator.

Reply:
@ye-luo I have to agree here. I'd say fixing the initialization argument from ions to srcs just restores the original behavior from ~2 weeks back. There the distance table didn't see the ions particleset until it was in the constructor body of LocalMomentEstimator so this didn't occur. So this should be merged.

The sanest test here would be to get to the point where we can watch for compiler warnings in the CI. Probably for both gcc and clang.